cmd/nomos/status/cluster_state.go

```diff
@@ -71,23 +71,28 @@ func unavailableCluster(ref string) *ClusterState {
 
 // RepoState represents the sync status of a single repo on a cluster.
 type RepoState struct {
-	scope      string
-	syncName   string
-	sourceType v1beta1.SourceType
-	git        *v1beta1.Git
-	oci        *v1beta1.Oci
-	helm       *v1beta1.HelmBase
-	status     string
-	commit     string
-	errors     []string
+	scope             string
+	syncName          string
+	sourceType        v1beta1.SourceType
+	git               *v1beta1.Git
+	oci               *v1beta1.Oci
+	helm              *v1beta1.HelmBase
+	status            string
+	commit            string
+	lastSyncTimestamp metav1.Time
+	errors            []string
 	// errorSummary summarizes the `errors` field.
 	errorSummary *v1beta1.ErrorSummary
 	resources    []resourceState
 }
 
 func (r *RepoState) printRows(writer io.Writer) {
 	fmt.Fprintf(writer, "%s%s:%s\t%s\t\n", util.Indent, r.scope, r.syncName, sourceString(r.sourceType, r.git, r.oci, r.helm))
-	fmt.Fprintf(writer, "%s%s\t%s\t\n", util.Indent, r.status, r.commit)
+	if r.status == syncedMsg {
```

Comment:
Should we always print the lastSyncTimestamp if it was synced before?

The current sync status might be ERROR, but we can still track the timestamp of last successful sync. This requires adding a `LastSyncedTimestamp` field in the status. The new field can be set along with the `LastSyncedCommit` field: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/api/configsync/v1beta1/sync_types.go#L36.

Reply:
If the sync status is not SYNCED, the users probably care more about why it is not synced. 

I don't feel comfortable to add new fields into the status APIs considering the feature complete date for 1.14 is coming soon. 

We can first deliver this PR to see whether this is good enough for the customers who want this feature. If not, we can explore the option of adding new status API fields.

Reply:
sounds good